### PR TITLE
Fix Condition when right operand is null (contains, startswith, endswith)

### PR DESCRIPTION
**Pull request by RodneyRichardson**
Prevents exceptions being thrown when right = null. Also adds test for when left operand is null, but these were handle correctly already.

Replaces stale #513.

### CODE REVIEW

File: src/DotLiquid/Condition.cs [hunk 1]
```diff
@@ -26,10 +26,10 @@ public class Condition
...
             { ">=", (left, right) => left != null && right != null && Comparer<object>.Default.Compare(left, Convert.ChangeType(right, left.GetType())) >= 0 },
-            { "contains", (left, right) => ((left is string) ? ((string)left).Contains((string)right) : (left is IEnumerable) ? Any((IEnumerable) left, (element) => element.BackCompatSafeTypeInsensitiveEqual(right)) : false) },
-            { "startsWith", (left, right) => (left is IList) ? EqualVariables(((IList) left).OfType<object>().FirstOrDefault(), right) : ((left is string) ? ((string)left).StartsWith((string) right) : false) },
-            { "endsWith", (left, right) => (left is IList) ? EqualVariables(((IList) left).OfType<object>().LastOrDefault(), right) : ((left is string) ? ((string)left).EndsWith((string) right) : false) },
-            { "hasKey", (left, right) => (left is IDictionary) ? ((IDictionary) left).Contains(right) : false },
+            { "contains", (left, right) => ((left is string && right != null) ? ((string)left).Contains((string)right) : (left is IEnumerable) ? Any((IEnumerable) left, (element) => element.BackCompatSafeTypeInsensitiveEqual(right)) : false) },
```
Comment (microalps) [inline src/DotLiquid/Condition.cs:29]: Besides rebasing, did you consider the below alternative?
```suggestion
            { "contains", (left, right) => ((left is string leftString) ? leftString.Contains((string)right ?? string.Empty) : (left is IEnumerable) ? Any((IEnumerable) left, (element) => element.BackCompatSafeTypeInsensitiveEqual(right)) : false) },

```

Emphasis on ` ?? string.Empty`. If you think about it, it doesn't mean the same thing, since string is also IEnumerable and will evaluate the second half if right is null. Same for the next few.

Comment (RodneyRichardson) [reply src/DotLiquid/Condition.cs:29]: I just tried this, and tests fail for `'bob' contains null`, and equivalent.

**Comment by RodneyRichardson** [comment]
Rebased and ready to go...

**Comment by microalps** [comment]
I've reviewed one more case and [confirmed](https://replit.com/@microalps/LiquidUnitTests#main.rb) that liquid returns TRUE for empty string on the right. Adding test cases and cleaning up some messages.

**Comment by RodneyRichardson** [comment]
Looks good. I fixed a variable name, and added some more tests.

I've noticed we're inconsistent with our string/numeric conversions in these - I've added tests to explicitly show the quirk. But it means `contains` will return false, but `endswith`/`startswith` might return true for the same arguments.

I don't propose we change this now, but maybe in a v3?

**Review by microalps — APPROVED** [review]
